packages/zone.js/lib/rxjs/rxjs.ts

```diff
@@ -50,22 +50,26 @@ type ZoneSubscriberContext = {
         },
         set: function(this: Observable<any>, subscribe: any) {
           (this as any)._zone = Zone.current;
-          (this as any)._zoneSubscribe = function(this: ZoneSubscriberContext) {
-            if (this._zone && this._zone !== Zone.current) {
-              const tearDown = this._zone.run(subscribe, this, arguments as any);
-              if (tearDown && typeof tearDown === 'function') {
-                const zone = this._zone;
-                return function(this: ZoneSubscriberContext) {
-                  if (zone !== Zone.current) {
-                    return zone.run(tearDown, this, arguments as any);
-                  }
-                  return tearDown.apply(this, arguments);
-                };
+          if (!subscribe) {
+            (this as any)._zoneSubscribe = subscribe;
+          } else {
+            (this as any)._zoneSubscribe = function(this: ZoneSubscriberContext) {
+              if (this._zone && this._zone !== Zone.current) {
+                const tearDown = this._zone.run(subscribe, this, arguments as any);
+                if (typeof tearDown === 'function') {
+                  const zone = this._zone;
+                  return function(this: ZoneSubscriberContext) {
+                    if (zone !== Zone.current) {
+                      return zone.run(tearDown, this, arguments as any);
+                    }
+                    return tearDown.apply(this, arguments);
+                  };
+                }
+                return tearDown;
               }
-              return tearDown;
-            }
-            return subscribe.apply(this, arguments);
-          };
+              return subscribe.apply(this, arguments);
```

Comment:
can we add `else` here. The early returns make this code confusing to read. 
```suggestion
             } else { return subscribe.apply(this, arguments); }
```

Reply:
yes, you are right, updated, thanks.